src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/CSharpTagHelperCodeRenderer.cs

```diff
@@ -340,18 +332,72 @@ private void RenderUnboundHTMLAttributes(IEnumerable<KeyValuePair<string, Chunk>
             }
         }
 
-        private void RenderTagHelperBody(IList<Chunk> children, bool bufferBody)
+        private void RenderTagHelperContent()
         {
-            // If we want to buffer the body we need to create a writing scope to capture the body content.
-            if (bufferBody)
+            // Rendering output is a runtime feature.
+            if (_designTimeMode)
             {
-                // Render all of the tag helper children in a buffered writing scope.
-                BuildBufferedWritingScope(children);
+                return;
             }
-            else
+
+            _writer.Write("if (")
+                   .Write(ExecutionContextVariableName)
+                   .Write(".")
+                   .Write(_tagHelperContext.ExecutionContextOutputPropertyName)
+                   .Write(".")
+                   .Write(_tagHelperContext.OutputContentSetPropertyName)
+                   .WriteLine(")");
+
+            // Render the body of the if statement, at this point in the codegen the TagHelperOutput's Content was set
```

Comment:
Maybe rephrase this statement. 

> At this point in the codegen, TagHelperOutput.Content is set. We need to use this to render the Content instead of executing the child content